task_manager/src/lib.rs

```diff
@@ -0,0 +1,279 @@
+mod select_all;
+
+use std::pin::pin;
+
+use crate::select_all::select_all;
+use futures::{future::LocalBoxFuture, Future, StreamExt};
+use tokio::signal;
+
+pub trait ManagedTask {
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> LocalBoxFuture<'static, anyhow::Result<()>>;
+}
+
+pub struct TaskManager {
+    tasks: Vec<Box<dyn ManagedTask>>,
+}
+
+pub struct TaskManagerBuilder {
+    tasks: Vec<Box<dyn ManagedTask>>,
+}
+
+pub struct StopableLocalFuture {
+    shutdown_trigger: triggered::Trigger,
+    future: LocalBoxFuture<'static, anyhow::Result<()>>,
+}
+
+impl Future for StopableLocalFuture {
+    type Output = anyhow::Result<()>;
+
+    fn poll(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Self::Output> {
+        pin!(&mut self.future).poll(cx)
+    }
+}
+
+impl<F, O> ManagedTask for F
+where
+    O: Future<Output = anyhow::Result<()>> + 'static,
+    F: FnOnce(triggered::Listener) -> O,
+{
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> LocalBoxFuture<'static, anyhow::Result<()>> {
+        Box::pin(self(shutdown))
+    }
+}
+
+impl Default for TaskManager {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl TaskManager {
+    pub fn new() -> Self {
+        Self { tasks: Vec::new() }
+    }
+
+    pub fn builder() -> TaskManagerBuilder {
+        TaskManagerBuilder { tasks: Vec::new() }
+    }
+
+    pub fn add(&mut self, task: impl ManagedTask + 'static) {
+        self.tasks.push(Box::new(task));
+    }
+
+    pub async fn start(self) -> anyhow::Result<()> {
+        let mut sigterm = signal::unix::signal(signal::unix::SignalKind::terminate()).unwrap();
```

Comment:
avoid the unwrap here since we're inside something that should be able to take `?` since it returns an anyhow result?